File: rn-expo/src/data/TmdbApiClient.ts
import axios, { AxiosInstance } from "axios";
import { MovieDetailDto, PopularMoviesResponseDto } from "./dtos";

export class TmdbApiClient {
  private client: AxiosInstance;

  constructor(baseUrl: string, apiKey: string) {
    this.client = axios.create({
      baseURL: baseUrl,
      params: { api_key: apiKey },
    });
  }

  async getPopularMovies(page: number): Promise<PopularMoviesResponseDto> {
    const response = await this.client.get<PopularMoviesResponseDto>(
      "/movie/popular",
      { params: { page } }
    );
    return response.data;
  }

  async getMovieDetail(id: number): Promise<MovieDetailDto> {
    const response = await this.client.get<MovieDetailDto>(`/movie/${id}`);
    return response.data;
  }
}

File: kmp-native/shared/src/commonMain/kotlin/com/stackbenchmark/kmpnative/data/remote/TmdbApiClient.kt
package com.stackbenchmark.kmpnative.data.remote

import com.stackbenchmark.kmpnative.data.dto.MovieDetailDto
import com.stackbenchmark.kmpnative.data.dto.PopularMoviesResponseDto
import io.ktor.client.HttpClient
import io.ktor.client.call.body
import io.ktor.client.request.get
import io.ktor.client.request.parameter

class TmdbApiClient(
    private val httpClient: HttpClient,
    private val baseUrl: String,
    private val apiKey: String
) {
    suspend fun getPopularMovies(page: Int): PopularMoviesResponseDto {
        return httpClient.get("$baseUrl/movie/popular") {
            parameter("api_key", apiKey)
            parameter("page", page)
        }.body()
    }

    suspend fun getMovieDetail(movieId: Int): MovieDetailDto {
        return httpClient.get("$baseUrl/movie/$movieId") {
            parameter("api_key", apiKey)
        }.body()
    }
}
